src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalescingAssignmentTests.cs

```diff
@@ -0,0 +1,1408 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests.Semantics
+{
+    [CompilerTrait(CompilerFeature.NullCoalescingAssignment)]
+    public class NullCoalescingAssignmentTests : CompilingTestBase
+    {
+        [Fact]
+        public void LocalLvalue()
```

Comment:
>LocalLvalue [](start = 20, length = 11)

I'd suggest adding a variation on this first test with a string type: `s ??= GetString()`.
The IL should be simpler since no wrapping into a `Nullable<T>`.